xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-resource/plugin.js

```diff
@@ -162,16 +162,51 @@
         },
         validate: function() {
           var resourceReference = this.getValue();
+          var resourceTypeConfig = $resource.types[resourceReference.type] || {};
           if (resourceReference.reference === '') {
             // Check if the selected resource type supports empty references.
-            var resourceTypeConfig = $resource.types[resourceReference.type] || {};
             if (resourceTypeConfig.allowEmptyReference !== true) {
               return this.getDialog().getParentEditor().localization.get('xwiki-resource.notSpecified',
                 this.getLabelElement().getText());
             }
+          } else if (resourceReference.notSelected && resourceTypeConfig.mustBeSelected) {
+            return this.validateInput(resourceReference);
           }
           return true;
         },
+        validateInput: function(resourceReference) {
+          if (!this.validationRequest) {
+            // Trigger a new validation.
+            this.validationRequest = this.validateAsync(resourceReference).always((function() {
+              // Re-submit the dialog after the current event is handled.
+              setTimeout((function() {
+                this.getDialog().click('ok');
+              }).bind(this), 0);
+            }).bind(this));
```

Comment:
You can use arrow / lambda functions to avoid the need for binding this (because they don't change the scope).

```suggestion
            this.validationRequest = this.validateAsync(resourceReference).always(() => {
              // Re-submit the dialog after the current event is handled.
              setTimeout(() => {
                this.getDialog().click('ok');
              }, 0);
            });
```